### PR TITLE
Add SaveCommand; Storage, ListStorage, ModuleListStorage, TaskListSto…

### LINKED ISSUE #77: Parser unit tests
Add unit tests for Parser, and move the current contents of ParserTest into ModHappyParserTest.

### PR DESCRIPTION
**Pull request by Ch40gRv1-Mu**
…rage, and Corresponding JUnit

Add: 1. Storage; 2. ListStorage; 3. ModuleListStorage; 4. TaskListStorage; 5. SaveCommand; 6. ParseTest; 7. ModuleListStorageTest; 8. TaskListStorageTest. Refactor Parse.java to ModHappyParse.java

### CODE REVIEW

File: src/main/java/seedu/duke/parsers/ModHappyParser.java [hunk 1]
```diff
@@ -69,6 +71,8 @@ private Parser getCommandParser(String commandWord) throws UnknownCommandExcepti
             return new DeleteParser();
         case (MARK_COMMAND_WORD):
             return new MarkParser();
+        case (SAVE_COMMAND_WORD):
+            return new NoArgumentParser(commandWord);
```
Comment (chooyikai) [inline src/main/java/seedu/duke/parsers/ModHappyParser.java]: Let's combine this with the fallthrough cases in lines 64-65.

File: src/main/java/seedu/duke/storage/ModuleListStorage.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+     */
+    @Override
+    public ArrayList<Module> jsonReader(String path) throws ModHappyException {
+        Gson gson = new GsonBuilder().create();
+        Path file = new File(path).toPath();
+        try {
+            Reader reader = Files.newBufferedReader(file, StandardCharsets.UTF_8);
+            Module[] list = gson.fromJson(reader, Module[].class);
+            ArrayList<Module> arrayList;
+            if (list != null) {
```
Comment (chooyikai) [inline src/main/java/seedu/duke/storage/ModuleListStorage.java]: Should we switch this to `!Objects.isNull(list)` instead for consistency with other places in the code?

File: src/main/java/seedu/duke/util/StringConstants.java [hunk 3]
```diff
@@ -55,6 +62,9 @@
     public static final String ERROR_PARSE_FAILED = "This parse failed 0_0";
     public static final String ERROR_UNKNOWN_COMMAND = "Sorry, I don't understand the following command:";
     public static final String ERROR_UNSUPPORTED_RESULT_TYPE = "Sorry, I don't understand the result format:";
+    public static final String ERROR_WRITE_STREAM_BROKEN = "Sorry, The write steams is broken";
+    public static final String ERROR_READ_STREAM_BROKEN = "Sorry, The read steams is broken";
```
Comment (chooyikai) [inline src/main/java/seedu/duke/util/StringConstants.java]: Minor typo, steams -> stream

**Review by chooyikai — COMMENTED** [review]
Just some minor comments, otherwise looks pretty good

File: src/main/java/seedu/duke/parsers/ModHappyParser.java (hunk 1 above)
Comment (Ch40gRv1-Mu) [reply src/main/java/seedu/duke/parsers/ModHappyParser.java]: okk

File: src/main/java/seedu/duke/storage/ModuleListStorage.java (hunk 2 above)
Comment (Ch40gRv1-Mu) [reply src/main/java/seedu/duke/storage/ModuleListStorage.java]: Updated

**Pull request merged**